### PR TITLE
app key renaming

### PR DESCRIPTION
**Pull request by spartan218**
Places where we need to rename accountID to application key ID moving forward.

Note: that account ID still persists in certain areas as it is needed for other b2 operations.

### CODE REVIEW

File: core/src/main/java/com/backblaze/b2/client/B2AccountAuthorizerSimpleImpl.java [hunk 1]
```diff
@@ -14,43 +14,43 @@
  * Almost everyone is expected to use this right now.
  */
 public class B2AccountAuthorizerSimpleImpl implements B2AccountAuthorizer {
-    private final String accountId;
+    private final String applicationKeyId;
```
Comment (certainmagic) [inline core/src/main/java/com/backblaze/b2/client/B2AccountAuthorizerSimpleImpl.java]: the claim is that changing this does NOT change the API nor does it change what we send on the wire, right?

File: core/src/main/java/com/backblaze/b2/client/credentialsSources/B2Credentials.java [hunk 2]
```diff
@@ -9,9 +9,9 @@
  */
 public interface B2Credentials {
     /**
-     * @return the accountId to use for b2_authorize_account.
+     * @return the applicationKeyId to use for b2_authorize_account.
      */
-    String getAccountId();
+    String getApplicationKeyId();
```
Comment (certainmagic) [inline core/src/main/java/com/backblaze/b2/client/credentialsSources/B2Credentials.java]: this is an incompatible API change, methinks.  with this change, i think we will need to bump the major version number of the SDK.  is that necessary?

File: core/src/main/java/com/backblaze/b2/client/credentialsSources/B2CredentialsFromEnvironmentSource.java [hunk 3]
```diff
@@ -8,7 +8,7 @@
 
 /**
  * This B2CredentialsSource reads the credentials from two environment variables:
- *   B2_ACCOUNT_ID
+ *   B2_APPLICATION_KEY_ID
```
Comment (certainmagic) [inline core/src/main/java/com/backblaze/b2/client/credentialsSources/B2CredentialsFromEnvironmentSource.java]: i think this change means that people who run programs which use this class will have their programs stop working until they change their environment variables.  is that necessary?

File: core/src/main/java/com/backblaze/b2/client/credentialsSources/B2Credentials.java (hunk 2 above)
Comment (spartan218) [reply core/src/main/java/com/backblaze/b2/client/credentialsSources/B2Credentials.java]: Can you help me understand why this is an incompatible API change?

File: core/src/main/java/com/backblaze/b2/client/credentialsSources/B2CredentialsFromEnvironmentSource.java (hunk 3 above)
Comment (spartan218) [reply core/src/main/java/com/backblaze/b2/client/credentialsSources/B2CredentialsFromEnvironmentSource.java]: If they upgrade to the newest SDK once this is merged I believe that they need to change their environment variables names, which is documented.

 I'm not sure how to answer your question about being necessary. The goal behind the app key renaming is to bring clarity to the name. My intention with this change was to bring b2-java-sdk users up to speed on the same terminology that we want everyone to use.

Comment (certainmagic) [reply core/src/main/java/com/backblaze/b2/client/credentialsSources/B2CredentialsFromEnvironmentSource.java]: briang & i talked and he's taking a stab at maintaining compatibility.

File: README.md [hunk 4]
```diff
@@ -54,7 +54,7 @@ SAMPLE
   * To run B2Sample, you will need to add your credentials to your environment
     in these environment variables:
 
-    * B2_APPLICATION_KEY_ID
+    * B2_ACCOUNT_ID OR B2_APPLICATION_KEY_ID
```
Comment (certainmagic) [inline README.md]: thanks.  :)

File: core/src/main/java/com/backblaze/b2/client/credentialsSources/B2CredentialsImpl.java [hunk 5]
```diff
@@ -9,19 +9,30 @@
  */
 public class B2CredentialsImpl implements B2Credentials {
     private final String accountId;
+    private final String appKeyId;
     private final String appKey;
 
     B2CredentialsImpl(String accountId,
                              String appKey) {
+
+        // for now consider any accountId an appKeyId
```
Comment (certainmagic) [inline core/src/main/java/com/backblaze/b2/client/credentialsSources/B2CredentialsImpl.java]: that *sounds* right.  let's talk it through together.

File: core/src/main/java/com/backblaze/b2/client/structures/B2AuthorizeAccountRequest.java [hunk 6]
```diff
@@ -10,25 +10,25 @@
...
+    public static Builder builder(String applicationKeyId,
                                   String applicationKey) {
-        return new Builder(accountId, applicationKey);
+        return new Builder(applicationKeyId, applicationKey);
 
     }
 
-    public String getAccountId() {
-        return accountId;
+    public String getApplicationKeyId() {
```
Comment (certainmagic) [inline core/src/main/java/com/backblaze/b2/client/structures/B2AuthorizeAccountRequest.java]: looks like we're losing getAccountId() here too.
this whole trying-to-be-compatible is awkward.

Comment (spartan218) [reply core/src/main/java/com/backblaze/b2/client/structures/B2AuthorizeAccountRequest.java]: yea I can add it back.

**Pull request closed without merging**